crates/ethereum/evm/src/execute.rs

```diff
@@ -118,16 +122,18 @@ where
 {
     /// Executes the transactions in the block and returns the receipts.
     ///
-    /// This applies the pre-execution changes, and executes the transactions.
+    /// This applies the pre-execution and post-execution changes that require an [EVM](Evm), and
+    /// executes the transactions.
     ///
     /// # Note
     ///
-    /// It does __not__ apply post-execution changes.
-    fn execute_pre_and_transactions<Ext, DB>(
+    /// It does __not__ apply post-execution changes that do not require an [EVM](Evm), for that see
+    /// [EthBlockExecutor::post_execution].
+    fn execute_state_transitions<Ext, DB>(
         &self,
         block: &BlockWithSenders,
-        mut evm: Evm<'_, Ext, &mut State<DB>>,
-    ) -> Result<(Vec<Receipt>, u64), BlockExecutionError>
+        evm: &mut Evm<'_, Ext, &mut State<DB>>,
```

Comment:
@mattsse is this OK to borrow mutably here

Reply:
no need